packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.ts

```diff
@@ -266,6 +266,17 @@ export default class UserStorageController extends BaseController<
     // We will remove this once the feature will be released
     isAccountSyncingEnabled: false,
     isAccountSyncingInProgress: false,
+    canSync: () => {
+      try {
+        this.#assertProfileSyncingEnabled();
+
+        return (
+          this.#accounts.isAccountSyncingEnabled && this.#auth.isAuthEnabled()
+        );
+      } catch {
+        return false;
+      }
```

Comment:
Yep fair.

The assert seems to just check state
https://github.com/MetaMask/core/blob/6cf64c648cb5c94aefd69348d1e219b320d2f51e/packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.ts#L638-L644

We might be able just create a safe one of these or copy the line.

Reply:
I think the try/catch is still valid, since not sure what `this.#auth.isAuthEnabled` would call under the hood or in the future (if it changes)

Reply:
`this.#auth.isAuthEnabled` checks the state of `AuthenticationController` in order to get `isSignedIn`.

I agree, I don't think it's 100% ideal to be honest, and we're again going to be checking UI-side if all of this is enabled. I'll look into it when I have a bit less on my plate to kind of centralize these checks.